code/datums/components/food/decomposition.dm

```diff
@@ -0,0 +1,64 @@
+//"Don't leave food on the floor, that's how we get ants"
+/datum/component/decomposition
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+	///How decomposed a specific food item is. This will go rather high due to the 2 second process timer.
+	var/decomposition_level = 0
+	var/is_stored = FALSE
+
+/datum/component/decomposition/Initialize()
+	if(!isatom(parent))
+		return COMPONENT_INCOMPATIBLE
+	RegisterSignal(parent, COMSIG_ITEM_DROPPED, .proc/table_check)
+	RegisterSignal(parent, COMSIG_ITEM_PICKUP, .proc/picked_up)
+	RegisterSignal(parent, COMSIG_STORAGE_ENTERED, .proc/storage_check) //Checks if you put it in storage
+	RegisterSignal(parent, COMSIG_CLOSET_CONTENTS_EXPOSED, .proc/table_check)
+	RegisterSignal(parent, COMSIG_TRY_STORAGE_HIDE_ALL, .proc/picked_up) //Checks if it's in a closed closet
+	RegisterSignal(parent, COMSIG_STORAGE_EXITED, .proc/table_check) //Checks if a storage object has been dumped
+	RegisterSignal(parent, COMSIG_PARENT_EXAMINE, .proc/examine)
+
+/datum/component/decomposition/Destroy()
+	UnregisterSignal(parent, COMSIG_ITEM_DROPPED, .proc/table_check)
```

Comment:
Move all of this (except the `STOP_PROCESSING`) to `UnregisterFromParent`. Then, make this a list in the form of:

```dm
UnregisterSignal(parent, list(
    COMSIG_ITEM_DROPPED,
    COMSIG_ITEM_PICKUP,
    // etc
))
```